include/flatbuffers/vector_downward.h

```diff
@@ -203,7 +207,11 @@ class vector_downward {
     memset(make_space(zero_pad_bytes), 0, zero_pad_bytes);
   }
 
-  void pop(size_t bytes_to_remove) { cur_ += bytes_to_remove; }
+  void pop(size_t bytes_to_remove) {
+    cur_ += bytes_to_remove;
```

Comment:
I presume you're checked that all places that modify `cur_` now also touch `size_` ?

Kind of a shame to have 2 vars like that, but it's worth it for that kind of speedup. I suppose if we removed `cur_` that would slow it down again..

Reply:
Yeah, i believed i checked all places, though I sort of relied on the flattests to catch any mistakes :)

I didn't analyze removing `cur_` and just keeping `size_`. I would have to take a look to see if that has an effect.

Reply:
> relied on the flattests

You too, like to live dangerously.

> I would have to take a look

Up to you.. it's likely slower, but it be awesome if it wasn't :)

Reply:
now that the `flatbuffers.h` is split into separate files, it might be nice to do small unit tests for each of the components. VectorDownward would be a good one to do first.

Reply:
Removing `cur_` and recalculating it when needed is slower than it is currently.